### ISSUE TITLE
Extraneous "missing field" error after parse error

### ISSUE THREAD

**Issue by alexcrichton** [rust-lang/rust#35400]
When compiling this code:

``` rust
struct A {
    a: u32,
}

fn foo(a: u32) -> A {
    A {
        a a, // note the missing colon
    }
}

fn main() {
    foo(1);
}
```

I get:

```
error: expected `:`, found `a`
 --> foo.rs:7:11
  |
7 |         a a, // note the missing colon
  |           ^

error[E0063]: missing field `a` in initializer of `A`
 --> foo.rs:6:5
  |
6 |     A {
  |     ^

error: aborting due to previous error
```

We probably shouldn't emit the second error!

cc @nrc

**Comment by estebank** [comment]
[We no longer](https://is.gd/mINKnq) emit the missing field error in 1.17:

```rust
error: expected one of `,` or `}`, found `a`
 --> <anon>:7:11
  |
7 |         a a, // note the missing colon
  |           ^
```
